### ISSUE TITLE
[Roberta] Contribute to an example using flask to serve Roberta models

### ISSUE THREAD

**Issue by neomatrix369** [pytorch/fairseq#1477]
I was wondering what you think about contributing an example that uses flask and exposes REST endpoints loading and allowing access to the different model types (small and large parameter spaces). (a) the endpoints would be accessible via curl or postman or any other rest service.

(b) In addition to giving curl access also serves a page that can be used to interact with the various Roberta models.

I think a simple example like this might be invaluable to newcomers to the site.

Let me know how best to go about with it.

**Comment by ngoyal2707** [comment]
@neomatrix369 While I think serving models through REST apis can be useful for some people, this is out of scope of `fairseq` library currently. We view `fairseq` as library to facilitate training and inference. 

You are definitely welcome to build such application using `fairseq` library and released models.

**Comment by neomatrix369** [comment]
> @neomatrix369 While I think serving models through REST apis can be useful for some people, this is out of scope of `fairseq` library currently. We view `fairseq` as library to facilitate training and inference.
> 
> You are definitely welcome to build such application using `fairseq` library and released models.

Thanks for the response, although I was suggesting this on two grounds:
- I would contribute the code (as per your coding guidelines)
- I found it useful to understand Roberta once I had the flask server working and the model responding

I'm not picking on the README page you have, but ready-made examples (with decent comments or docs/READMEs) are easier for developers of all levels to pick-up and run with than reading docs and following them. I experienced this during the hackathon at the FB office in London.

I do understand that's not what this library/package/framework is meant for but an example shows it in a better light, in my view.
